lib/platform/github/index.ts

```diff
@@ -1735,3 +1744,79 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
   }
   return alerts;
 }
+
+async function pushFiles({
+  branchName,
+  files,
+  message,
+}: CommitFilesConfig): Promise<CommitSha | null> {
+  const additions: FileAddition[] = [];
+  const deletions: FileDeletion[] = [];
+
+  for (const file of files) {
+    if (file.type === 'addition') {
+      additions.push(file);
+    } else {
+      deletions.push(file);
+    }
+  }
+
+  const executableFiles = additions.filter(({ isExecutable }) => isExecutable);
+  if (executableFiles.length) {
+    logger.warn(
+      { branchName, executableFiles: executableFiles.map(({ path }) => path) },
+      'Platform-native commit: found executable files'
+    );
+    return null;
+  }
+
+  const fileChanges = {
+    additions: additions.map(({ path, contents }) => ({
+      path,
+      contents: Buffer.from(contents).toString('base64'),
+    })),
+    deletions: deletions.map(({ path }) => ({ path })),
+  };
+
+  const variables = {
+    repo: config.repository,
+    repositoryId: config.repositoryId,
+    branchName: `refs/heads/${branchName}`,
+    oid: config.defaultBranchOid,
+    fileChanges,
+    message,
+  };
+
+  try {
+    const { data, errors } = await githubApi.requestGraphql<{
+      createCommitOnBranch: { commit: { oid: string } };
+    }>(commitFilesMutation, { variables });
+
+    if (errors) {
+      logger.warn(
+        { branchName, errors },
+        'Platform-native commit: GraphQL errors'
+      );
+      return null;
+    }
+
+    return data?.createCommitOnBranch?.commit?.oid;
+  } catch (err) {
+    logger.debug({ branchName, err }, 'Platform-native commit: unknown error');
+    return null;
+  }
+}
+
+export async function commitFiles(
+  config: CommitFilesConfig
+): Promise<CommitSha | null> {
+  const commitResult = await git.prepareCommit(config);
+  if (!commitResult) {
+    return null;
+  }
```

Comment:
Is there sufficient logging from git for this scenario or should we add something here saying platform commit aborted?

Reply:
It has enough logging, I believe

Reply:
Added some log message anyway